app/models/move.rb

```diff
@@ -171,6 +171,15 @@ def for_feed
     feed_attributes
   end
 
+  def determine_supplier
+    # Supplier choice is based on from_location and effective date. As the from location cannot change after the
+    # move is created we only need to look for date changes - if no change then skip the supplier update for speed.
+    return supplier unless date_changed? || date_from_changed?
```

Comment:
I think this is working for new moves because they must have either a date or a from_date, and so `date_changed? || date_from_changed?` evaluates to `true`. But, it might be a little more readable to have instead:
```
return supplier unless new_record? || date_changed? || date_from_changed?
```